### PR TITLE
Add dwarfism and gigantism traits, vehicle disability aids

### LINKED ISSUE #70369: Players can no longer "fit" into wheelchairs and drive them
### Describe the bug

You can no longer sit in and drive a wheelchair

### Attach save file

N/A

### Steps to reproduce

1: Select a profession that starts your character with a wheelchair.
2: Try to ride the wheelchair

### Expected behavior

I expected my character to fit into the wheelchair.

### Screenshots

_No response_

### Versions and configuration

Latest Experimental. 2023-12-22-0719

### Additional context

_No response_

### LINKED ISSUE #70502: I can't fit through a broken car door
### Describe the bug

I cannot traverse a broken car door.

### Attach save file

Save file: [Apollo-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13790161/Apollo-trimmed.tar.gz)

Non standard mods:
[mods.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13790164/mods.tar.gz)

### Steps to reproduce

1. Load attached save file
2. Try to go through the door due south of you
3. There's not enough room for you to fit there.

### Expected behavior

I can travel through a broken door as I can other broken vehicle parts.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3803 (22H2)
- Game Version: 2a65f10 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Fewer Soldier Extras [Fewer_Soldiers],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Inglonias Scenarios and Professions [Inglonias_Scenarios]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by worm-girl**
#### Summary
Features "Adds extreme height chargen traits, adds tools for disabled drivers"

#### Purpose of change

#70184 made height interaction with vehicles more dynamic. This naturally suggests doing more with the system. This PR also fixes a few leftover bugs from that one.

#### Describe the solution

- Adds very tall and very short traits at chargen. The heights expressed by these traits (88cm to 250cm) are certainly extreme, but within the bounds of real-life human expression. These allow you to start in the small or large size category, but don't introduce stat changes like the existing mutations do - people like Jason Acuña or Yao Ming are exceptional, but very different from a guy who is turning into a bear or a mouse.
- These traits do adjust your stomach size, calorie requirements, and the amount of weight you can carry. The very tall trait will cause high blood pressure ( #70239 ) once that's possible, but that can wait for another PR.
- Very Tall and Very Short change to Large and Small respectively, so if you're a mod creator getting pinged for review it's because I had to copy the new traits to your mods.
- Adds new flags to standard vehicle controls, NEED_LEG and INOPERABLE_SMALL. You need at least one working leg to drive a car and you can't do it if you're too short to reach the pedals. These flags are currently only being applied to the standard vehicle controls item, no need to worry about bicycles or anything.
- Adds hand controls for vehicles ( https://youtu.be/tkdWr_Ie0Ec? ). These will let you drive if you're small or tiny, or have broken or missing legs, and are pretty easy to install and remove.
- Adds pedal extenders and makeshift pedal extenders for vehicles. These are just rubber (or wooden) pads that you clamp (or duct tape) to the pedals so you can drive even if you're short.
- Adds the CARGO_PASSABLE flag to a few storage spaces that needed it. Fixes #70369 . I have left the flag off of the garbage can because it's a huge 350ml plastic bin and probably shouldn't be boardable at all, but that's an issue for another day.
- Adds pedal extenders and hand controls to hospitals and (much more rarely) garages, cars, automotive stores, big box stores, etc.
- Adjusts a comment in the aisle json to suggest that it's about .5m and not 1m square for parity with other vehicle parts. See comments below #70184 . This doesn't affect anything but may prevent confusion in the future.
- Placed a couple of pedal extenders and hand controls around in the world. These are quite rare items IRL, especially pedal extenders. You can occasionally find both in mechanic shops, and every once in a while hand controls will appear in a dump or a recycling center.
- Added recipes for all 3 mobility aid items. None are very hard to make, though the hand controls do require a bit of welding.
- fixes #70502 
- Sneaks in a small fix for an issue where grab+pushing appliances and vehicles was giving you the cramped space effect
- Slightly upgrades capacity of car seats so that large people can (uncomfortably) fit in reclining and bench seats. They cannot fit in non-reclining seats unless the spot is not enclosed

#### Describe alternatives you've considered

It might be tempting to add bonuses or maluses to very tall and very short on top of what's there, as conditions like gigantism can come with all kinds of complications, but given the wide diversity of causes for extreme height in real life, and the freeform nature of our chargen system I think it's best to leave that in the hands of the player. As such I've left the terms vague (very short and very tall vs gigantism or what have you) and given them minimal knock-on effects.

I'll also be leaving very short and very tall out of the pool of random traits for NPCs, as I believe the incidence of these would be too small to be correctly modeled by our random NPC trait system. It'd be weird if every fiftieth guy you ran into was a little person. I suggest we add a static NPC or two to the world to show off these traits if we'd like them represented.

#### Testing

- Crafted, installed, uninstalled, and reinstalled all the car parts
- Tested that each car part works fine if you don't need it but there happens to be one installed
- Tested that the parts work if you do need them
- Tested that you can't drive if you don't
- Tested this on controls other than "vehicle controls". Saw that these mobility aids (designed for gas/brake pedals only) don't matter for things like sails or bicycles
- Saw that small characters can still ride bicycles (kids ride bikes all the time!)
- Mutated back and forth to the existing small and large mutations, with and without very tall and very short as starting traits. Worked fine
- Tested with remote vehicle controls to ensure there was no conflict
- Tested AFS, Magiclysm, Dinomod mutations to make sure they still work
- Tested 12/29 bugfixes, including new seat size, travois and wooden box flags, and preventing APPLIANCE and OBSTACLE parts from improperly squishing the player when pushed

#### Additional context

image

IDEAS FOR FUTURE PRS: Maybe doctor's offices should have mobility aids? But they'd probably need a specific place to go and their own item group. There are also stores that sell just these kinds of things. It might also be fun to make electronic vehicle controls that use a video game controller or something.

### CODE REVIEW

File: data/json/items/vehicle/utilities.json [hunk 1]
```diff
@@ -287,5 +287,53 @@
...
+    "volume": "740 ml",
+    "category": "veh_parts",
+    "price": 10000,
+    "price_postapoc": 40,
+    "melee_damage": { "bash": 3 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "pedal_extenders_makeshift",
+    "name": { "str": "hand controls" },
```
Comment (TheShadowFerret) [inline data/json/items/vehicle/utilities.json:326]:
```suggestion
    "name": { "str": "makeshift pedal extenders" },
```

**Comment by fairyarmadillo** [comment]
Alright, I think we're all set. The remaining test failures are the same old overmap stuff, not related to this PR.

**Comment by LyleSY** [comment]
Love it, thanks for patching the mods as part of this

**Comment by kasanryukin** [comment]
So this is visible since it's related to your vehicle changes:  A filled non-cargo tile (like an isle, door, or a seat) becomes impassable when full.
image

It pretty much renders the extra storage those tiles normally allow to be useless in large vehicles with a traversable interior space.

**Comment by fairyarmadillo** [comment]
That"s not true of doors, and is very much the point for seats and aisles. You are making a decision about whether to store things there or have it be a passable space.

**Comment by kasanryukin** [comment]
With the two height traits, are they a requirement of selecting the trait, or can chargen be altered so that heights with-in those ranges automatically apply the trait?  For example, setting your character's height between 88cm and 120cm would assigned you the very short trait?

**Comment by fairyarmadillo** [comment]
> With the two height traits, are they a requirement of selecting the trait, or can chargen be altered so that heights with-in those ranges automatically apply the trait? For example, setting your character's height between 88cm and 120cm would assigned you the very short trait?

You must pick the trait. Your height is bounded as before from 145 to 200cm, and the trait is applied on top of that. Your new height is determined relative to the initial height you chose. This is how the Large and Little traits already work in the game, you just can't normally select them from chargen.

**Comment by Maleclypse** [comment]
`Adds new flags to standard vehicle controls, NEED_LEG and INOPERABLE_SMALL. You need at least one working leg to drive a car and you can't do it if you're too short to reach the pedals. These flags are currently only being applied to the standard vehicle controls item, no need to worry about bicycles or anything.`
Very cool, so are you thinking about doing anything with Gastropod foot preventing or needing alternative bicycle pedals in the future?  I need to get back to body parting the rest of the limb mutations sometime.

**Comment by fairyarmadillo** [comment]
> `Adds new flags to standard vehicle controls, NEED_LEG and INOPERABLE_SMALL. You need at least one working leg to drive a car and you can't do it if you're too short to reach the pedals. These flags are currently only being applied to the standard vehicle controls item, no need to worry about bicycles or anything.` Very cool, so are you thinking about doing anything with Gastropod foot preventing or needing alternative bicycle pedals in the future? I need to get back to body parting the rest of the limb mutations sometime.

With bicycles and gastropod foot, I left the old solution in place. That means for now, you can't use your gastropod foot to ride a bike, but you can use it to operate the gas and brake pedals in a car, since that only needs one working "leg". I'm not sure that's appropriate (it might work? I could picture someone doing it if they had like a snake tail instead of legs, but a gastropod foot is a bit more unwieldy) but I couldn't think of one elegant solution that fit all cases. Definitely open to suggestions here.

I also want to come back later and make pedal extenders not work for tiny characters. I also think there should be a way for a huge character to custom-build a vehicle seat they can fit in, but I'm not totally certain how that would work. Maybe it would just need to be a super heavy part, or we could create a high roof which could also be used for things like semi trailers?

**Comment by kasanryukin** [comment]
For too short characters, as well as paraplegics, there are specialized
parts that essentially create a motorcycle like driving experience, which I
wouldn't mind doing the research and json legwork in an addendum PR. For
huge characters, much like those over 7ft, you'd need a custom seat to
accommodate such a large frame. You already have checks for passability in
seats and aisles with too much inventory, so in theory you can just add a
check for the traits and disallow those, and require a special seat. The
vehicle system can already check for installed components, and while you
can't really enforce it with out changing all the frames, requiring the
special seat to only attach to heavy frames should be viable.  If you go
that route, Korg might want to add said flags to the orichalcum frame as
well.

**Comment by fairyarmadillo** [comment]
That's essentially what the hand controls are.

As for huge characters, it's not simply the size of the seat or the strength of the frame. There is simply no possible way a grizzly bear could drive a car designed for humans because every part of the car is too small for it. You couldn't put a bear-sized seat in a sedan. It's also not just a matter of frames because you couldn't put one in a humvee either. The entire vehicle, or at least that part of it, would need to be designed for the animal to fit in.

**Comment by titusvalid** [comment]
200 CM lanky guy no mutations cant get through windows, surely thats not intended

**Comment by worm-girl** [comment]
> 200 CM lanky guy no mutations cant get through windows, surely thats not intended

It is.

**Pull request merged**